### ISSUE TITLE
Location Channel Buffer Issue

### ISSUE THREAD

**Issue by Nailik** [icerockdev/moko-geo#30]
Hi

i found a problem with the Channel design.

Currently everytime there is a new location it gets send to the channel

```
 trackerScope.launch {
            extendedLocationsChannel.send(extendedLocation)
            locationsChannel.send(latLng)
        }
```

But when the app starts tracking but watches only one flow (e.g. only `getLocationsFlow()`),
there is never a receive action on the second channel (`extendedLocationsChannel`)

This leds to the issue that  `locationsChannel.send(latLng)` is not called after 64 locations where send because the channels are implemented as:

```
private val locationsChannel = Channel<LatLng>(Channel.BUFFERED)
private val extendedLocationsChannel = Channel<ExtendedLocation>(Channel.BUFFERED)
```

And in the docs they mention 

```

        /**
         * Requests a buffered channel with the default buffer capacity in the `Channel(...)` factory function.
         * The default capacity for a channel that [suspends][BufferOverflow.SUSPEND] on overflow
         * is 64 and can be overridden by setting [DEFAULT_BUFFER_PROPERTY_NAME] on JVM.
         * For non-suspending channels, a buffer of capacity 1 is used.
         */
        public const val BUFFERED: Int = -2
```

so this should be changed to 

```
        /**
         * Requests a conflated channel in the `Channel(...)` factory function. This is a shortcut to creating
         * a channel with [`onBufferOverflow = DROP_OLDEST`][BufferOverflow.DROP_OLDEST].
         */
        public const val CONFLATED: Int = -1
```

**Comment by Alex009** [comment]
released in 0.5.0

**Comment by madamczewski** [comment]
@Alex009 Seems like the issue has been only fixed on Android.
I encounter the same issue on iOS that after 1-5 minutes I don't get location updates anymore.
I guess the buffer in the `https://github.com/icerockdev/moko-geo/blob/master/geo/src/iosMain/kotlin/dev/icerock/moko/geo/LocationTracker.kt` has to be changed to Conflated as well.

btw is there any reason to use Channels instead of SharedFlow?
Seems like using SharedFlow is easier + there is a potential bug with Channels:
When the LocationTracker is reused (e.g. singleton) and you call the `getLocationsFlow()` two times or more, then only one Flow gets updated because Channels can emit only to single receiver.
